x-pack/plugins/apm/public/selectors/latency_chart_selectors.ts

```diff
@@ -35,15 +41,28 @@ export function getLatencyChartSelector({
       anomalyTimeseries: undefined,
     };
   }
+
+  const environmentForMlJob =
+    selectedEnvironment === ENVIRONMENT_ALL.value &&
+    availableEnvironments?.length === 1
+      ? availableEnvironments?.[0]
+      : selectedEnvironment;
```

Comment:
I don't think we should handle environment for ML differently than anything else. 
What about redirecting to the first environment if there is only one?

Reply:
That means when the user goes back to the services overview the environment selected will be stuck to the environment we redirected to. IIRC we discussed that before and decided we didn't want that, but I might be wrong.

Reply:
True. 
Thing is, won't this all be solved in 7.12 when we remove "All" environment, and default to "production"?

It seems a little risky to do this change this late in 7.11 - especially since it's not a regression (it's been like this since 7.10).

Reply:
Is that decision final for 7.12? I'm fine with postponing. IMHO it's a pretty terrible experience right now but it's not technically wrong and we can document it.